spec.html

```diff
@@ -8631,18 +8631,58 @@ <h1>WithBaseObject ( )</h1>
       <emu-clause id="sec-object-environment-records">
         <h1>Object Environment Records</h1>
         <p>Each <dfn>object Environment Record</dfn> is associated with an object called its <em>binding object</em>. An object Environment Record binds the set of string identifier names that directly correspond to the property names of its binding object. Property keys that are not strings in the form of an |IdentifierName| are not included in the set of bound identifiers. Both own and inherited properties are included in the set regardless of the setting of their [[Enumerable]] attribute. Because properties can be dynamically added and deleted from objects, the set of identifiers bound by an object Environment Record may potentially change as a side-effect of any operation that adds or deletes properties. Any bindings that are created as a result of such a side-effect are considered to be a mutable binding even if the Writable attribute of the corresponding property has the value *false*. Immutable bindings do not exist for object Environment Records.</p>
-        <p>Object Environment Records created for `with` statements (<emu-xref href="#sec-with-statement"></emu-xref>) can provide their binding object as an implicit *this* value for use in function calls. The capability is controlled by a _withEnvironment_ Boolean value that is associated with each object Environment Record. By default, the value of _withEnvironment_ is *false* for any object Environment Record.</p>
+        <p>Object Environment Records created for `with` statements (<emu-xref href="#sec-with-statement"></emu-xref>) can provide their binding object as an implicit *this* value for use in function calls. The capability is controlled by a Boolean [[WithEnvironmentFlag]] field.</p>
+        <p>Object Environment Records have the additional state fields listed in <emu-xref href="#table-additional-fields-of-object-environment-records"></emu-xref>.</p>
+        <emu-table id="table-additional-fields-of-object-environment-records" caption="Additional Fields of Object Environment Records">
+          <table>
+            <tbody>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[BindingObject]]
+              </td>
+              <td>
+                Object
+              </td>
+              <td>
+                The binding object of this Environment Record.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[WithEnvironmentFlag]]
+              </td>
+              <td>
+                Boolean
+              </td>
+              <td>
+                Determines whether this Environment Record provides its binding object as an implicit *this* value for use in function calls.
```

Comment:
This should also mention that when this field is **true**, then properties specified in:
> [Get][sec-get]\(<var>[objectEnvironmentRecord][sec-object-environment-records]</var>.[[BindingObject]], [%Symbol.unscopables%][sec-well-known-symbols])

are excluded from:
> <var>[objectEnvironmentRecord][sec-object-environment-records]</var>.[HasBinding][sec-object-environment-records-hasbinding](…)

[sec-get]: https://tc39.es/ecma262/#sec-get-o-p
[sec-well-known-symbols]: https://tc39.es/ecma262/#sec-well-known-symbols
[sec-object-environment-records]: https://tc39.es/ecma262/#sec-object-environment-records
[sec-object-environment-records-hasbinding]: https://tc39.es/ecma262/#sec-object-environment-records-hasbinding-n

Reply:
I've changed this line to "Indicates whether this Environment Record is created for a `with` statement.", focusing on what causes the field to be true, rather than the effects it has, thus avoiding the problem of stating one effect but not another.